eclair-node/src/main/scala/fr/acinq/eclair/Boot.scala

```diff
@@ -68,12 +72,21 @@ object Boot extends App with Logging {
     if (config.getBoolean("api.enabled")) {
       logger.info(s"json API enabled on port=${config.getInt("api.port")}")
       val apiPassword = config.getString("api.password") match {
-        case "" => throw EmptyAPIPasswordException
-        case valid => valid
+        case "" => None
+        case valid => Some(valid)
+      }
+      val apiCookiePassword = if (config.getBoolean("api.cookie-enabled")) {
+        Some(generateCookie(config.getString("api.cookie-path"), config.getBoolean("api.cookie-group-readable")))
+      } else {
+        None
+      }
+      if (apiPassword.isEmpty && apiCookiePassword.isEmpty) {
+        throw new RuntimeException("json-rpc api requires that either password or cookie is enabled")
```

Comment:
Instead of an unnamed exception, you should replace the `EmptyAPIPasswordException` in `Setup.scala` with a `APICredentialsNotConfiguredException`